[fix] Don't use adapted args when calling Service.apply

samschlegel opened the pull request:
This fixes #277

Currently the generated code for FinagleService utilizes adapted args when calling `finagle.Service#apply`. For many people this generates a warning due to `-Ywarn-adapted-args`, and for some (ourselves included) causes the compilation to fail due to `-Yno-adapted-args`

To solve, we explicitly wrap arguments in a 2-tuple

As an aside, and this is a contentious topic, but I think it might be a good idea to add `-Yno-adapted-args` to the build scripts for scrooge, so that issues like this don't crop up in the future. It's a fairly common flag, and it's annoying to have to conditionally disable just for scrooge generated code. Though it will be removed in Scala 3 (scala/scala#6505), so might auto-tupling!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/twitter/scrooge?pullRequest=292) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/twitter/scrooge?pullRequest=292) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/twitter/scrooge?pullRequest=292) it.</sub>

mosesn commented:
This seems reasonable, can you change the commit message to match the template we suggest in CONTRIBUTING.md? https://github.com/twitter/scrooge/blob/develop/CONTRIBUTING.md#pull-requests

mosesn commented:
@samschlegel whichever you prefer.  just lmk which you decide to do.

yufangong commented:
Hey @samschlegel, can we merge this PR first? Or you plan to update it with build flag?

yufangong commented on scrooge-generator/src/main/resources/scalagen/finagleService.mustache:
```diff
@@ -115,7 +115,7 @@ class {{ServiceName}}$FinagleService(
       service match {
         case _root_.scala.Some(svc) =>
-          svc(iprot, msg.seqid)
+          svc((iprot, msg.seqid))
```
Second thought, I believe you need to also make this change to GoldService$FinagleService.scala and PlatinumService$FinagleService.scala. Can you run the test to confirm?

samschlegel replied:
Only needed to add to GoldService :smile:

samschlegel commented:
Ah apologies, this got away from me. I will try to take a look at this again this week and get the tests working. I'll also make an issue for adding the build flag, so that someone else can tackle that if they have time before I do

mosesn commented:
Thanks @samschlegel! Have you had a chance to take another look at this?

samschlegel commented:
I've updated the generated files, and added the `-Yno-adapted-args` flag to build.sbt so we can avoid running into this in the future. I need to play with SBT_OPTS in order to get the full build running on my machine, but I'll update any more cases of adapted args as I find them

kevinoliver commented:
@samschlegel this looks good to me. i'll try to get it merged.

kevinoliver commented:
Looks like there is still some issues in the sbt build which I see when I pull this down.

```
[error] /data/jenkins/workspace/source_3/scrooge/scrooge-generator-tests/src/test/scala/com/twitter/scrooge/finagle_integration/ScalaIntegrationTest.scala:41:52: not enough arguments for method ===: (rightSide: Any)(implicit equality: org.scalactic.Equality[Unit])Boolean.
[error] Unspecified value parameter rightSide.
[error]     assert(await(muxClient.setDuck(20L, "McDuck")) === ())
[error]                                                    ^
[error] /data/jenkins/workspace/source_3/scrooge/scrooge-generator-tests/src/test/scala/com/twitter/scrooge/finagle_integration/ScalaIntegrationTest.scala:234:40: not enough arguments for method ===: (rightSide: Any)(implicit equality: org.scalactic.Equality[Unit])Boolean.
[error] Unspecified value parameter rightSide.
[error]     assert(await(clnt.setDuck(3, "x")) === ())
[error]
```

mosesn commented:
@samschlegel looks like your build is still failing in travisci.  I'm restarting the build to see if it was just travisci flakiness.

mosesn commented:
Nope, looks like you have a failing test: 
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 375, Failed 1, Errors 0, Passed 374
[error] Failed tests:
[error] 	com.twitter.scrooge.goldfile.ScalaGoldFileTest
```

samschlegel commented:
Odd, it seemed to have worked locally. Will retry.

kevinoliver commented:
Thanks again @samschlegel, this got merged in a0a2320d43ab1ffcfd1f5fe50c009486b0982d5a
